### PR TITLE
Remove libgc, which duplicates bdw-gc

### PR DESCRIPTION
**Pull request by fxcoudert**
Fixes issue #12984 

`libgc` is more recent, not used as dependency, and less complete, so it should go. But let's move its test to `bdw-gc`.

### CODE REVIEW

File: Formula/bdw-gc.rb [hunk 1]
```diff
@@ -31,4 +31,30 @@ def install
...
+          int **p = (int **) GC_MALLOC(sizeof(int *));
+          int *q = (int *) GC_MALLOC_ATOMIC(sizeof(int));
+          assert(*p == 0);
+          *p = (int *) GC_REALLOC(q, 2 * sizeof(int));
+        }
+        return 0;
+      }
+    EOS
+
+    system ENV.cc, "-lgc", "-o", "test", "test.c"
```
Comment (ilovezfs) [inline Formula/bdw-gc.rb]: Let's add `"-I#{include}"` and `"-L#{lib}"`

**Comment by fxcoudert** [comment]
Upstream website certificate expired, reported there https://github.com/ivmai/bdwgc/issues/157 (and by email to Hans-J. Boehm), waiting until it's back up to test and merge that.

**Comment by MikeMcQuaid** [comment]
Nice work! Can we add an alias to match the old name?

**Comment by ilovezfs** [comment]
Alias is not the same as formula rename. I'm not sure we want a formula rename.

**Comment by fxcoudert** [comment]
Given that `libgc` is recent, is not a dependency for anything, and represents 0.4% of the installs that `bdw-gc` has… I think we should have neither.

**Comment by MikeMcQuaid** [comment]
I think an alias is sufficient too.

**Comment by fxcoudert** [comment]
Hans Boehm indicated in email that HTTPS support on his website was not deliberate and is not supported. Moving back to HTTP.

**Comment by fxcoudert** [comment]
@BrewTestBot test this pleast

**Comment by ilovezfs** [comment]
The symlink file name should be libgc not libgc.rb

**Pull request merged**